doc/source/f2py/f2py-examples.rst

```diff
@@ -0,0 +1,326 @@
+
+Creating source for a basic extension module
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Consider the following subroutine, contained in a file named :file:`add.f`
+
+.. code-block:: fortran
+
+    C
+          SUBROUTINE ZADD(A,B,C,N)
+    C
+          DOUBLE COMPLEX A(*)
+          DOUBLE COMPLEX B(*)
+          DOUBLE COMPLEX C(*)
+          INTEGER N
+          DO 20 J = 1, N
+             C(J) = A(J)+B(J)
+     20   CONTINUE
+          END
+
+This routine simply adds the elements in two contiguous arrays and places the
+result in a third. The memory for all three arrays must be provided by the
+calling routine. A very basic interface to this routine can be automatically
+generated by f2py::
+
+    f2py -m add add.f
+
+This command will produce an extension module named :file:`addmodule.c` in the
+current directory. This extension module can now be compiled and used from
+Python just like any other extension module.
+
+Creating a compiled extension module
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can also get f2py to both compile :file:`add.f` along with the produced
+extension module leaving only a shared-library extension file that can
+be imported from Python::
+
+    f2py -c -m add add.f
+
+This command leaves a file named add.{ext} in the current directory
+(where {ext} is the appropriate extension for a Python extension
+module on your platform --- so, pyd, *etc.* ). This module may then be
+imported from Python. It will contain a method for each subroutine in
+add (zadd, cadd, dadd, sadd). The docstring of each method contains
+information about how the module method may be called::
+
+    >>> import add
+    >>> print(add.zadd.__doc__)
+    zadd(a,b,c,n)
+
+    Wrapper for ``zadd``.
+
+    Parameters
+    ----------
+    a : input rank-1 array('D') with bounds (*)
+    b : input rank-1 array('D') with bounds (*)
+    c : input rank-1 array('D') with bounds (*)
+    n : input int
+
+Improving the basic interface
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The default interface is a very literal translation of the Fortran
+code into Python. The Fortran array arguments must now be NumPy arrays
+and the integer argument should be an integer. The interface will
+attempt to convert all arguments to their required types (and shapes)
+and issue an error if unsuccessful. However, because it knows nothing
+about the semantics of the arguments (such that C is an output and n
+should really match the array sizes), it is possible to abuse this
+function in ways that can cause Python to crash. For example::
+
+    >>> add.zadd([1, 2, 3], [1, 2], [3, 4], 1000)
+
+will cause a program crash on most systems. Under the covers, the
+lists are being converted to proper arrays but then the underlying add
+loop is told to cycle way beyond the borders of the allocated memory.
+
+In order to improve the interface, directives should be provided. This
+is accomplished by constructing an interface definition file. It is
```

Comment:
I am not familiar with Chasm - can you suggest a rewording if you think this is confusing? Maybe just call it signature file instead of interface definition file?